source/common/unified_memory_allocation/include/uma/memory_pool_ops.h

```diff
@@ -0,0 +1,51 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ */
+
+#ifndef UMA_MEMORY_POOL_OPS_H
+#define UMA_MEMORY_POOL_OPS_H 1
+
+#include <uma/base.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/// This structure comprises function pointers used by corresponding  umaPool*
+/// calls. Each memory pool implementation should initialize all function pointers.
+struct uma_memory_pool_ops_t {
+  /// Version of the ops structure.
+  /// Should be initialized using UMA_VERSION_CURRENT
+  uint32_t version;
```

Comment:
A client that uses UMA might be compiled with a different version of UMA. This field allows UMA to understand which version of ops structure is used by the client to support ABI compatibility.

Reply:
And this is specifically to allow for future changes in the `ops` structure, like adding new fields or removing existing ones? In such a case, `ops` created with an older UMA version will result in a failure when creating a new memory pool, right?

Reply:
yeah, the version field is introduced to support compatibility with previous versions of the UMA in case ops structure is modified. Regarding what should be done in case of version mismatch, if ops structure is just extended with some field and we can use some default behavior than we do not need to return the error. But in general you are right, based on version field we need to implement some logic in case of version mismatch.

Reply:
There is also another implication of this. We can either bump UMA version **only** when providing significant changes in ops structure, ie. ones that makes the old ops unusable or we can check for version range (this would be cumbersome).

Reply:
Yeah, it depends on the meaning of the `version` field in the ops structure.

- If it contains the version of the library, then the library version is increased on each release and we should check for the because ops structure might stay unchanged between versions).
- If the `version` field contains the version of the ops structure itself then the version is increased only when the ops structure is changed.